metashare/api/models.py

```diff
@@ -76,28 +79,50 @@ def full_org_type(self):
     @property
     def instance_url(self):
         try:
-            return self.social_account.extra_data["instance_url"]
+            return self.salesforce_account.extra_data["instance_url"]
         except (AttributeError, KeyError):
             return None
 
     @property
     def token(self):
-        account = self.social_account
+        account = self.salesforce_account
         if account and account.socialtoken_set.exists():
-            token = self.social_account.socialtoken_set.first()
-            return (fernet_decrypt(token.token), fernet_decrypt(token.token_secret))
+            token = self.salesforce_account.socialtoken_set.first()
+            try:
+                return (
+                    fernet_decrypt(token.token) if token.token else None,
+                    token.token_secret if token.token_secret else None,
+                )
+            except InvalidToken:
+                return (None, None)
         return (None, None)
 
     @property
-    def social_account(self):
-        return self.socialaccount_set.first()
+    def salesforce_account(self):
+        return self.socialaccount_set.filter(provider="salesforce-production").first()
```

Comment:
@wlonk Ah, this explains why the `salesforce-custom` login isn't working currently.

Reply:
So we want to ensure that you only ever have one linked SF account at a time, from any provider, right? And we'll force disconnect the others. In which case this can be `provider__startswith="salesforce-"` and it'll work.